dropshot/examples/colliding-error-type-names.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 Oxide Computer Companyy
+
+//! Example of an API using a user-defined error type to serialize error
+//! representations differently from [`dropshot::HttpError`].
+
+use dropshot::endpoint;
+use dropshot::ApiDescription;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
+use dropshot::RequestContext;
+use dropshot::ServerBuilder;
+
+mod latex {
+    use super::*;
+
+    /// Errors returned by pdflatex.
+    ///
+    /// Good luck figuring out what these mean!
```

Comment:
This might just be me but when reading this I found myself confused about what any of this stuff is before I realized that these are just opaque tokens you might know if you'd fought `pdflatex` recently.  Take it or leave it but it might be clearer if the example came from the space that dropshot consumers are likely to know already.

Reply:
Sure, I'm happy to change the example, or change the docs to make it clearer that I just picked random error strings from my memory, as the errors themselves were supposed to be context-free. I'd be happy to contrive an example that's actually doing some kind of (basic) validation, or something; I started by sketching out examples mostly as a way to ensure that code using the new APIs compiled, and I think it's not a bad idea to go back and make them more useful for actually learning how to use the APIs.